torchtune/modules/kv_cache.py

```diff
@@ -40,14 +40,20 @@ def __init__(
         self.register_buffer(
             "v_cache", torch.zeros(cache_shape, dtype=dtype), persistent=False
         )
-        self.size = 0
+        self.register_buffer(
+            "cache_pos", torch.arange(0, cache_shape[2]), persistent=False
+        )
         self.batch_size = batch_size
 
     def reset(self) -> None:
         """Reset the cache to zero."""
         self.k_cache.zero_()
         self.v_cache.zero_()
-        self.size = 0
+        self.cache_pos = torch.arange(0, self.k_cache.shape[2])
+
+    @property
+    def size(self) -> int:
+        return self.cache_pos[0].item()
```

Comment:
as a rule of thumb, its never good to call item. It is not performant and has issues with torch export

Reply:
Noob q, how should I be doing this?